pkgs/applications/graphics/processing/default.nix

```diff
@@ -1,73 +1,94 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, fetchurl, xmlstarlet, makeWrapper, ant, jdk, rsync, javaPackages, libXxf86vm, gsettings-desktop-schemas }:
+{ lib, stdenv, fetchFromGitHub, fetchurl, ant, unzip, makeWrapper, jdk11, javaPackages, rsync, gsettings-desktop-schemas, xorg }:
+let
+  vaqua = fetchurl {
+    name = "VAqua9.jar";
+    url = "https://violetlib.org/release/vaqua/9/VAqua9.jar";
+    sha256 = "cd0b82df8e7434c902ec873364bf3e9a3e6bef8b59cbf42433130d71bf1a779c";
+  };
+
+  jna = fetchurl {
+    name = "jna-5.8.0.zip";
+    url = "https://github.com/java-native-access/jna/archive/5.8.0.zip";
+    sha256 = "7cRkkxyo2Ld19Nif8oWceITt7dYMUnQlrs0R5e0AmPE=";
+  };
+
+  gluon = fetchurl {
+    name = "javafx-16-sdk-linux.zip";
+    url = "https://gluonhq.com/download/javafx-16-sdk-linux";
+    sha256 = "W9Q5LQ0tak3cEr7fjZEtSvF/hiLwVg2+AoH1bE+e0Uw=";
+  };
+
+  batik = fetchurl {
+    name = "batik-bin-1.14.zip";
+    url = "https://archive.apache.org/dist/xmlgraphics/batik/binaries/batik-bin-1.14.zip";
+    sha256 = "vNYa4jW7QSPucGHKFWmLI2eTOnRYfmzDNzUiwmzCX28=";
+  };
 
+  ffmpeg = fetchurl {
+    url = "https://download.processing.org/ffmpeg/4.4/ffmpeg-linux-amd64.gz";
+    name = "ffmpeg-4.4.gz";
+    sha256 = "45M5oArfpTuGONyeem9+a2bXCrfw7TEILoZ4znXrcPo=";
+  };
```

Comment:
is that their own fork of ffmpeg? if its just a normal ffmpeg, add ffmpeg to arguments and copy `ffmpeg.src`